src/Compilers/CSharp/Portable/Compilation/AwaitExpressionInfo.cs

```diff
@@ -41,6 +41,10 @@ public bool Equals(AwaitExpressionInfo other)
 
         public override int GetHashCode()
         {
+            if (_awaitableInfo is null)
+            {
+                return 0;
+            }
```

Comment:
Suggestion: while you're here could change the `Equals` implementation to use pattern matching and avoid the double cast.